docs/src/modules/spring/pages/actions-as-controller.adoc

```diff
@@ -1 +1,112 @@
+<2> Expose the command handler as a `POST` endpoint at specified path.
+<3> Check if the added item is carrots.
+<4> If it is "carrots" immediately return an error, disallowing adding the item.
+<5> For allowed requests, use `kalixClient` to get a `DeferredCall`.
+<6> The `DeferredCall` is used with `effects().forward()` to forward the request to the entity.
+
+NOTE: You might be wondering what the `kalixClient` is about. For now, think of it as a lightweight HTTP client allowing you to reach out to other Kalix services. All details can be found at xref:call-another-service.adoc[] chapter.
+
+== Transform Request and Response to Another Component
+
+The `asyncReply` and `asyncEffect` effects allow us to process and transform a request before calling another component and then also transform the response.
+
+As an example, let us look at the problem of creating a new entity with an id generated by the service itself.
+
+In this example we implement an `initializeCart` command for the controller Action which returns the generated id that can subsequently be used to interact with the cart.
+
+.src/main/java/com/example/api/ShoppingCartController.java
+[source,java]
+----
+include::example$spring-valueentity-shopping-cart/src/main/java/com/example/api/ShoppingCartController.java[tag=initialize]
+----
+<1> Generate a new UUID.
+<2> Use the `kalixClient` to create a `DeferredCall` for `create` on the shopping cart -  note the use of the full path, empty body and the expected reply type `ShoppingCartDTO`.
+<3> `execute()` on the `DeferredCall` immediately triggers a call and returns a `CompletionStage` for the response.
+<4> Once the call succeeds or fails the `CompletionStage` is completed or failed, we can transform the result from `CompletionStage<Empty>`
+to `CompletionStage<Effect<String>>` using `handle`.
+<5> On a successful response, we create a reply effect passing back the `cartId`.
+<6> If the call leads to an error, we create an error effect asking the client to retry.
+<7> `effects().asyncEffect()` allows us to reply with a `CompletionStage<Effect<String>>`.
+
+The action generates a link:https://en.wikipedia.org/wiki/Universally_unique_identifier[UUID] to use as entity id for the shopping cart. UUIDs are extremely unlikely to lead to the same id
+being generated, but to completely guarantee two calls can never be assigned the same shopping cart we
+make use of the "boundary of consistency" provided by the entity - the entity will only process a single command at a time
+and can safely make decisions based on its state - for example to only allow creation once by storing something in its state
+signifying that it has been created.
+
+In this case we mark that the entity has been created using a creation timestamp in the shopping cart state stored on first
+`create` call (when the timestamp has the default value of `0`). If the cart has already been stored with a timestamp we return an error effect:
```

Comment:
```suggestion
`create` call - the timestamp has the default value of `0`. If the cart has already been stored with a timestamp we return an error effect:
```

Reply:
for the record, I did this change but also for the java counterpart